client/settings/payment-methods-selector/test/index.js

```diff
@@ -2,106 +2,200 @@
-
-		userEvent.click( giroPay.getByRole( 'checkbox' ) );
-		userEvent.click( sofort.getByRole( 'checkbox' ) );
-
-		fireEvent.click( getByText( 'Add selected' ) );
+		user.click(
+			getByRole( 'button', {
+				name: 'Cancel',
+			} )
+		);
 
-		expect( addSelectedPaymentMethods ).toHaveBeenCalledTimes( 1 );
-		expect( addSelectedPaymentMethods ).toHaveBeenCalledWith( [
-			'woocommerce_payments_giropay',
-			'woocommerce_payments_sofort',
-		] );
+		expect(
+			useEnabledPaymentMethodIds().updateEnabledPaymentMethodIds
+		).not.toHaveBeenCalled();
 
-		expect( handleClose ).toHaveBeenCalledTimes( 1 );
+		expect(
+			queryByRole( 'button', {
+				name: 'Cancel',
+			} )
+		).toBeNull();
 	} );
-
-	test( 'allows to preselect payment methods', () => {
-		const { getAllByRole } = render(
-			<PaymentMethodsSelector
-				enabledPaymentMethods={ [ 'woocommerce_payments_giropay' ] }
-			/>
+	test( 'Only payment methods that are not enabled are listed', () => {
+		useEnabledPaymentMethodIds.mockReturnValue( {
+			enabledPaymentMethodIds: [
+				'woocommerce_payments',
+				'woocommerce_payments_sofort',
+			],
+			updateEnabledPaymentMethodIds: jest.fn( () => {} ),
+		} );
+
+		const { getByRole, getAllByRole, queryByRole } = render(
+			<PaymentMethodsSelector />
 		);
 
+		const addPaymentMethodButton = getByRole( 'button', {
+			name: 'Add payment method',
+		} );
+		user.click( addPaymentMethodButton );
+
 		const paymentMethods = getAllByRole( 'listitem' );
-		expect( paymentMethods ).toHaveLength( 3 );
+		expect( paymentMethods ).toHaveLength( 2 );
+		expect( queryByRole( 'checkbox', { name: 'Sofort' } ) ).toBeNull();
+	} );
+	test( 'Selecting payment methods does not update enabled payment methods', () => {
+		useEnabledPaymentMethodIds.mockReturnValue( {
+			enabledPaymentMethodIds: [
+				'woocommerce_payments',
+				'woocommerce_payments_sepa',
+			],
+			updateEnabledPaymentMethodIds: jest.fn( () => {} ),
+		} );
+
+		const { getByRole, getAllByRole } = render(
+			<PaymentMethodsSelector />
+		);
 
-		const giroPay = within( paymentMethods[ 0 ] );
-		expect( giroPay.getByRole( 'checkbox' ).checked ).toBeTruthy();
+		const addPaymentMethodButton = getByRole( 'button', {
+			name: 'Add payment method',
+		} );
+		user.click( addPaymentMethodButton );
 
-		const sofort = within( paymentMethods[ 1 ] );
-		expect( sofort.getByRole( 'checkbox' ).checked ).toBeFalsy();
+		const paymentMethods = getAllByRole( 'listitem' );
+		const paymentMethodCheckbox = within( paymentMethods[ 0 ] ).getByRole(
+			'checkbox'
+		);
+		user.click( paymentMethodCheckbox );
 
-		const sepa = within( paymentMethods[ 2 ] );
-		expect( sepa.getByRole( 'checkbox' ).checked ).toBeFalsy();
+		expect( paymentMethodCheckbox ).toBeChecked();
+		expect(
+			useEnabledPaymentMethodIds().updateEnabledPaymentMethodIds
+		).not.toHaveBeenCalled();
+	} );
+	test( 'Selecting a payment method and clicking "Add selected" adds the method and closes the modal', () => {
```

Comment:
```suggestion
	} );

	test( 'Selecting a payment method and clicking "Add selected" adds the method and closes the modal', () => {
```

(minor) some breathing room between tests could help with readability (also in the previous lines)

Reply:
Done!